### PR TITLE
Git arc hacking

### PR DESCRIPTION
**Pull request by bsdjhb**
I haven't tested any of this yet, but wanted to at least give you a sense of what I'm hacking on.

### CODE REVIEW

**Comment by bsdjhb** [comment]
I want to see if there are more things to move under verbose by adding >3 redirects.  I'll have to create some reviews and exercise some more commands for that.  git arc stage (which is the only thing I've tested so far) seems fine in terms of verbosity.  I suspect arc diff --create won't be something we can easily shut up (and I'm not sure we want to).

File: git-arc [hunk 1]
```diff
@@ -203,33 +231,62 @@ restore_head()
...
+    fi
+
+    orig=$(save_head)
+    prev=""
+    for commit in ${commits}; do
+        if create_one_review "$commit" "$reviewers" "$subscribers" "$prev" \
+                             "$doprompt"; then
+            prev=$(commit2diff "$commit")
+        else
+            prev=""
```
Comment (bsdjhb) [inline git-arc:342]: Removing this else would instead stitch together the commits that aren't rejected by the user into a single stack.  I'm not sure what semantics we want (should the user rejecting a diff just terminate entirely, or just skip that revision, and if so, should skipping create a "hole" and a new stack like this patch does, or still create a single stack)

**Comment by bsdjhb** [comment]
I was able to test 'git arc create -l' a bit today and fixed a few nits.  I had to add -no-patch to actually get the compact display I wanted.  I also explicitly list them one at a time as otherwise duplicates get filtered (e.g. if I used a single `git show --oneline --no-patch ${commits}`).  Listing them one at a time does mean that --no-pager is required to not be super annoying.

The fact you have to exit the pager before getting the yes/no prompt for individual commits in non-list mode might not be obvious to users either, so we might want to default to no-pager there and/or add a config variable to control if the pager gets used for reviewing individual commits.

**Comment by bsdjhb** [comment]
Sample output for `git arc list` after the change to honor Phab URLs:

```
b67e44075539 No Review      : Add comment explaining Git commit message hook
672784750001 closed         D27583: Don't try to adjust a TLS TOE socket that has been closed.
878e208f4930 Needs Review   D27833: Restructure the crypto(7) manpage and add authentication algorithms.
85c4acea6ffd Accepted       D27834: Reject AES-GCM and AES-CCM sessions with per-op keys.
4c5f8ac91e73 Needs Review   D27835: Tidy some crypto-related lines in sys/conf/files.
2558588538b0 Accepted       D27836: Add an OCF algorithm for ChaCha20-Poly1035 AEAD.
6cba4f7fdf5a Needs Review   D27837: Add an implementation of CHACHA20_POLY1035 to cryptosoft.
005fdb4c2477 Needs Review   D27838: Add Chacha20-Poly1305 AEAD coverage.
8cc684d2aabf Needs Review   D27839: Add Chacha20-Poly1305 as a KTLS cipher suite.
7532272d218c Needs Review   D27841: Add Chacha20-Poly1305 support in the OCF backend for KTLS.
```

The first two commits are in upstream `main` and already committed and thus not visible in `arc list`, hence the lowercase "closed" status for one with a valid Phab URL.  The rest are in a patch series in review with the patches in various states.   The manual diff2status doesn't do the coloring of `arc list` (which doesn't show up here).

**Review by Danger2099 — APPROVED** [review]

**Pull request merged**